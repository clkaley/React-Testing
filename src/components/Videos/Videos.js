import React from 'react'
import '../../../src/App.css';


const Videos = ({videos=[ ]}) => {

    if(videos.length<=0){
        return <h1>Sorry, no videos item found </h1>
    }

  return (
    <div className='App'>
        {videos.map((video,i)=>
            <h3 key={i}>{video}</h3>
        )}
    </div>
  )
}

export default Videos