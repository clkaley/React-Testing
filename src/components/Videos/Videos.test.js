import React from "react";
import  renderer from "react-test-renderer";
import Videos from './Videos'


test('Videos Empty', ()=>{
    //dom-tree oluÅŸturuyoruz.
    const tree=renderer.create(<Videos />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Videos 1 Items Correctly', ()=>{
    const tree=renderer.create(<Videos videos={['jest']}/>);
    expect(tree).toMatchSnapshot();
});



test('Videos All Items Correctly', ()=>{
    const tree=renderer.create(<Videos videos={["youtube","udemy","tutor"]}/>);
    expect(tree).toMatchSnapshot();
});