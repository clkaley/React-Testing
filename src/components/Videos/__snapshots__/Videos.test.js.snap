// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Videos 1 Items Correctly 1`] = `
<div
  className="App"
>
  <h3>
    jest
  </h3>
</div>
`;

exports[`Videos All Items Correctly 1`] = `
<div
  className="App"
>
  <h3>
    youtube
  </h3>
  <h3>
    udemy
  </h3>
  <h3>
    tutor
  </h3>
</div>
`;

exports[`Videos Empty 1`] = `
<h1>
  Sorry, no videos item found 
</h1>
`;
